src/storage/kv/store.rs

```diff
@@ -97,15 +173,31 @@ pub struct Core {
     /// Commit log for store.
     pub(crate) clog: Arc<RwLock<Aol>>,
     /// Manifest for store to track Store state.
-    pub(crate) manifest: Aol,
+    pub(crate) manifest: RwLock<Aol>,
     /// Transaction ID Oracle for store.
     pub(crate) oracle: Arc<Oracle>,
     /// Value cache for store.
-    /// The assumption for this cache is that it could be useful for
+    /// The assumption for this cache is that it should be useful for
     /// storing offsets that are frequently accessed (especially in
     /// the case of range scans)
     pub(crate) value_cache: Arc<RwLock<LruCache<u64, Bytes>>>,
```

Comment:
In Surreal core, we recently switched from LruCache to QuickCache. Async + concurrent

https://crates.io/crates/quick_cache

Reply:
Thanks for the suggestion @emmanuel-keller !

Reply:
I am using the LRU cache and planning to replace it with the s3-fifo cache for better hit ratios. Will try to benchmark this wrt QuickCache

Reply:
Also added QuickCache here @emmanuel-keller !